packages/flame/lib/src/components/input/joystick_component.dart

```diff
@@ -45,7 +45,7 @@ class JoystickComponent extends HudMarginComponent with Draggable {
   late Vector2 _baseKnobPosition;
 
   JoystickComponent({
-    required this.knob,
+    PositionComponent? knob,
```

Comment:
Line 87 can fail now:
https://github.com/flame-engine/flame/pull/1099/files#diff-6e372118ee33f5a48d17030bcf2ec58a7dee48383c3bb2c8077a2a9f5a854a57R87
That line should be moved to `onMount`, similarly to how it is done here:
https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/components/input/hud_button_component.dart#L46-L55

Actually everything in `onLoad` can be moved to `onMount`.

Reply:
So generally I made it similar to `HudButtonComponent`. It makes more sense to have both `knob` and `background` as nullable and assert these in `onMount`.